.editorconfig

```diff
@@ -0,0 +1,170 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+indent_style = space
+insert_final_newline = true
+max_line_length = 99
+tab_width = 4
+trim_trailing_whitespace = true
+ij_continuation_indent_size = 8
+# ij_formatter_off_tag = @formatter:off
+# ij_formatter_on_tag = @formatter:on
+# ij_formatter_tags_enabled = false
+# ij_smart_tabs = false
+
+# For docstrings and comments.
+ij_visual_guides = 72
+
+# ij_wrap_on_typing = false
+
+[{*.h,*.hpp,*.cpp}]
+indent_size = 2
+tab_width = 2
+ij_continuation_indent_size = 4
+
+[*.feature]
+indent_size = 2
+# ij_gherkin_keep_indents_on_empty_lines = false
+
+[*.proto]
+indent_size = 2
+tab_width = 2
+# ij_continuation_indent_size = 4
+# ij_protobuf_keep_blank_lines_in_code = 2
+# ij_protobuf_keep_indents_on_empty_lines = false
+# ij_protobuf_keep_line_breaks = true
+# ij_protobuf_space_after_comma = true
+# ij_protobuf_space_before_comma = false
+# ij_protobuf_spaces_around_assignment_operators = true
+# ij_protobuf_spaces_within_braces = false
+# ij_protobuf_spaces_within_brackets = false
+
+[.editorconfig]
+# ij_editorconfig_align_group_field_declarations = false
+# ij_editorconfig_space_after_colon = false
+# ij_editorconfig_space_after_comma = true
+# ij_editorconfig_space_before_colon = false
+# ij_editorconfig_space_before_comma = false
+# ij_editorconfig_spaces_around_assignment_operators = true
+
+[*.json]
+indent_size = 2
+# ij_json_keep_blank_lines_in_code = 0
+# ij_json_keep_indents_on_empty_lines = false
+# ij_json_keep_line_breaks = true
+# ij_json_space_after_colon = true
+# ij_json_space_after_comma = true
+# ij_json_space_before_colon = true
+# ij_json_space_before_comma = false
+# ij_json_spaces_within_braces = false
+# ij_json_spaces_within_brackets = false
+# ij_json_wrap_long_lines = false
+
+[*.yml]
+indent_size = 2
+
+[*.dox]
+max_line_length = 72
+
+[*.md]
+max_line_length = 72
+# ij_markdown_force_one_space_after_blockquote_symbol = true
+# ij_markdown_force_one_space_after_header_symbol = true
+# ij_markdown_force_one_space_after_list_bullet = true
+# ij_markdown_force_one_space_between_words = true
+# ij_markdown_keep_indents_on_empty_lines = false
+# ij_markdown_max_lines_around_block_elements = 1
+# ij_markdown_max_lines_around_header = 1
+# ij_markdown_max_lines_between_paragraphs = 1
+# ij_markdown_min_lines_around_block_elements = 1
+# ij_markdown_min_lines_around_header = 1
+# ij_markdown_min_lines_between_paragraphs = 1
+
+[{*.pb,*.textproto}]
+indent_size = 2
+tab_width = 2
+ij_continuation_indent_size = 4
+# ij_prototext_keep_blank_lines_in_code = 2
+# ij_prototext_keep_indents_on_empty_lines = false
+# ij_prototext_keep_line_breaks = true
+# ij_prototext_space_after_colon = true
+# ij_prototext_space_after_comma = true
+# ij_prototext_space_before_colon = false
+# ij_prototext_space_before_comma = false
+# ij_prototext_spaces_within_braces = true
+# ij_prototext_spaces_within_brackets = false
```

Comment:
[consider] Removing all the file types we're highly unlikely to ever have in the project?

Reply:
Simplified significantly in 804ca886